pkg/reconciler/pipelinerun/resources/pipelineref_test.go

```diff
@@ -334,13 +357,15 @@ func TestGetPipelineFunc_RemoteResolution(t *testing.T) {
 	}}
 	for _, tc := range testcases {
 		t.Run(tc.name, func(t *testing.T) {
-			resolved := test.NewResolvedResource([]byte(tc.pipelineYAML), nil /* annotations */, sampleRefSource.DeepCopy(), nil /* data error */)
+			resolved := test.NewResolvedResource([]byte(tc.pipelineYAML), nil /* annotations */, v1beta1SampleRefSource.DeepCopy(), nil /* data error */)
```

Comment:
👋  one of the remaining questions I got is for the RefSource here to work, shall I swap all the v1beta1 RefSource to v1 from https://github.com/tektoncd/pipeline/blob/8d1a90346918c71fa628db8d50fefb759079ddd1/test/resolution.go#L46? cc @Yongxuanzhang @chuangw6

Reply:
to add more detail, meaning we need to change [refSource version in pkg/resolution/common/interface.go](https://github.com/tektoncd/pipeline/blob/8d1a90346918c71fa628db8d50fefb759079ddd1/pkg/resolution/common/interface.go#L62) to v1?

Reply:
I think we need to. I don't see a reason why we don't swap it as well

Reply:
Looking at the resolver interface, I think this would actually be a breaking change: https://github.com/tektoncd/pipeline/blob/main/docs/how-to-write-a-resolver.md#the-resolve-method

Since all resolvers must implement the Resolve method, which returns a ResolvedResource, changing the type of a ResolvedResource field would mean all resolvers must update their code. (Since the v1beta1 and v1 versions are identical, I don't think it's a big difference, but if we can swap the storage version of the api without making breaking changes, we should.)

Reply:
Actually, on second thought, I think it's OK to swap out the types in the resolver interface. Existing resolvers do not need to update as a result of this change, since they operate on the resolutionrequest CRD. They'd only need to update if pulling in a new version of pipelines for some reason.